src/simd/hook.cc

```diff
@@ -124,6 +128,22 @@ fvec_hook(std::string& simd_type) {
         simd_type = "GENERIC";
     }
 #endif
+
+#if defined(__ARM_NEON)
+    fvec_inner_product = fvec_inner_product_neon;
```

Comment:
Just curios, is there any context why we are using raw functions with different names instead of a class with template?

Reply:
template cannot work on the code runtime, template need choosed on the compiling time.